library/Faktory/Client.hs

```diff
@@ -172,16 +179,14 @@ sendUnsafe Settings {..} conn cmd args = do
 --
 -- Do not use outside of @'withMVar'@, this is not threadsafe.
 --
-recvUnsafe :: Settings -> Connection -> IO (Maybe ByteString)
+recvUnsafe :: Settings -> Connection -> IO (Either String (Maybe ByteString))
```

Comment:
What is "unsafe" about this? Typically `unsafe` is reserved for things that are memory unsafe. I'm assuming it was originally called unsafe because it `hush`ed the `Left`. Does it need this label now?

Reply:
I had the same confusion. See the note at end of commit message / pull request description, or haddock on this function.

Happy to rename, however you like. You named it first time around.

Reply:
Maybe `recvUnlocked` or `recvNonReentrant`? You could also do the `ST` trick and have a single function that eliminates it, e.g.:

```haskell
newtype Conn s = Conn { unConn :: Connection }

withConn :: Client -> (forall s. Conn s -> m a) -> m a
withConn Client{..} body = withMVar clientConnection $ body . Conn

recv :: Settings -> Conn s -> IO (Maybe ByteString)
recv ...

withConn client $ \conn -> do
  bytes <- recv conn
  whatever bytes
```

Though I doubt it's really worth the boilerplate for something internal

Reply:
You can leave it for now, but I'm likely going to explore the `ST` trick that @cdparks mentioned. It is more explicit and typesafe, which I like.